app/models/user.rb

```diff
@@ -65,6 +66,10 @@ def update_user_forms
 
 private
 
+  def requires_name?
+    name_was.present? || role_changed?(to: :editor)
```

Comment:
Can I double check this?

So we want to make `name` required if the user already had a name or if their role is being upgraded to :editor?

```suggestion
    name_was.present? || editor?
```
This would make it required to always have a name if the user previously had a name set or if they are actually an editor. I dont think we need to check if their role is changing to make it required.

Reply:
Thanks for flagging this up... it made me realise it should be `role_changed?(from: :trial)` 😅

Reply:
I guess we could also do `name_was.present? || editor? || super_admin?`, or `name_was.present? || !trial?`. Do you have a preference from those three options?

Reply:
I think the last option makes the most sense and clear that all users should have the name except if they are a trial user

Reply:
Okay, I've changed this and the other use of `role_changed` in a new commit: https://github.com/alphagov/forms-admin/pull/665/commits/2a83c29dfa3db3bc932a328a70208d6066c5f0c7